src/plugins/dashboard/public/dashboard_app/no_data/dashboard_app_no_data.tsx

```diff
@@ -122,6 +118,7 @@ export const DashboardAppNoDataPage = ({
             },
             path: '#/create',
           });
+          showPage();
```

Comment:
Why was this added here? 🤔

Reply:
There is a bug on main and I almost broke the fix out into a seperate PR. Here is the commit that added this https://github.com/elastic/kibana/pull/194892/commits/380356464be72fe027d36a8b9f4bfb02a5d81b43. A failing test pointed this out, but I forgot which one. Not sure why it started failing in this branch, but its flaky. Running it locally, the test only failed 20% or runs.

The problem is that `onTryESQL` callback does not call `showPage` and thus continues to display no data page.

Steps to see bug on main
1) install web logs sample data
2) delete data view and dashboard that ship with sample data
3) open dashboard application
4) change URL from `list` to `create`. Click `Try ESQL` button. Notice that application gets stuck and does not show dashboard with ESQL.

Reply:
One more update on this. 

I ended up reverting `showPage` call since it did not fix the root cause of the flaky test.

https://github.com/elastic/kibana/pull/194892/commits/4d813720c5d7a64c899fb445cd84a3f686964cb0 fixed the flaky test. The problem was that the new dashboard loading is much faster so `createOptions.getIncomingEmbeddable` was getting called before `isDashboardAppInNoDataState`. This caused  that `isDashboardAppInNoDataState` to return true when it should have returned false because `createOptions.getIncomingEmbeddable` cleared the incoming embeddable before `isDashboardAppInNoDataState` tried to read from it. Now `DashboardApp` reads incoming embeddable in a single place and uses that where needed to remove the race condition. 

This means that the bug where being on the `/dashboard/create` page and clicking `Try ESQL` still exists and users will get stuck on `/dashboard/create` in the no data state. But this is a bug on main and should be addressed separately. I will open an issue to track.